test/api/test_arrow.cpp

```diff
@@ -1,133 +0,0 @@
-#include "catch.hpp"
```

Comment:
Why was this deleted without replacement?

Reply:
Yes, these tests were a bit annoying because any changes on the ArrowArrayStream Interface would have to be duplicated there. 
This also just tests the API right, has nothing to do with arrow itself.
so I felt that this is sufficiently covered by the python tests.